markets/storageadapter/dealpublisher.go

```diff
@@ -321,26 +321,36 @@ func (p *DealPublisher) publishReady(ready []*pendingDeal) {
 	}
 
 	// Validate each deal to make sure it can be published
-	validated := make([]*pendingDeal, 0, len(ready))
-	deals := make([]market.ClientDealProposal, 0, len(ready))
+	validatedByProvider := make(map[address.Address][]*pendingDeal, 0)
```

Comment:
I prefer the current name. It is clear from the type that this is a map by provider.